src/mbgl/style/sources/geojson_source_impl.cpp

```diff
@@ -132,6 +138,7 @@ void GeoJSONSource::Impl::loadDescription(FileSource& fileSource) {
 
             loaded = true;
             observer->onSourceLoaded(base);
+            req.reset();
```

Comment:
Yeah, it should be done in `setURL`. Resetting here will disable automatic refreshes based on the caching headers, which is necessary to support things like traffic layers.